database/seeders/UsersIdiomasSeeder.php

```diff
@@ -13,35 +14,35 @@ class UsersIdiomasSeeder extends Seeder
      */
     public function run(): void
     {
-        DB::table('user_idiomas')->truncate();
-
-
-        DB::table('user_idiomas')->insert([
-            [
-                'user_id' => 1,
-                'idioma_id' => 1,
-                'nivel' => 'Básico',
-                'certificado' => 0,
-            ],
-            [
-                
-                'user_id' => 1,
-                'idioma_id' => 2,
-                'nivel' => 'Intermedio',
-                'certificado' => 1,
-            ],
-            [
-                'user_id' => 2,
-                'idioma_id' => 1,
-                'nivel' => 'Intermedio',
-                'certificado' => 0,
-            ],
-            [
-                'user_id' => 2,
-                'idioma_id' => 2,
-                'nivel' => 'Avanzado',
-                'certificado' => 1,
-            ],
-        ]);
+        UsersIdiomas::truncate();
+
+        $users = User::all();
+        $idiomas = Idiomas::all();
+        $niveles = ["Básico", "Intermedio", "Avanzado"];
+
+        foreach ($users as $user) {
+
+            $numIdiomas = rand(1, $idiomas->count());
+
+            $idiomasAsignados = [];
+
+            for ($i = 0; $i < $numIdiomas; $i++) {
+
+                do {
+                    $idiomaIdSeleccionado = $idiomas[rand(0, $idiomas->count() - 1)]->id;
+                } while (in_array($idiomaIdSeleccionado, $idiomasAsignados));
+
+                $idioma = new UsersIdiomas();
```

Comment:
Si decidieras mantener el modelo `UsersIdiomas`, podrías utilizarlo para definir las relaciones de `User` con `Idiomas`, como se explica [aquí](https://laravel.com/docs/10.x/eloquent-relationships#defining-custom-intermediate-table-models).

En cambio, si decidieras eliminar el modelo `UsersIdiomas`, deberías utilizar el método [attach](https://laravel.com/docs/10.x/eloquent-relationships#updating-many-to-many-relationships)